countwords/main.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"fmt"
+	"unicode"
+)
+
+func IsSep(r rune) bool { return unicode.IsSpace(r) || unicode.IsPunct(r) || unicode.IsSymbol(r) }
+
+func CountOccurrences(s string) (words []string, occurrences []uint) {
+	var tmp []rune
+	wasSep := true
+	for _, r := range s {
+		if IsSep(r) {
+			if !wasSep {
+				s = append(s, string(tmp))
+				tmp = nil
+			}
+			wasSep = true
+		} else {
+			wasSep = false
+			tmp = append(tmp, r)
+		}
+	}
+	if len(tmp) > 0 {
+		s = append(s, string(tmp))
+	}
+	return
+}
```

Comment:
Did you copy it from `words` function that we wrote during the lesson? Why wouldn't you just call it (if you need it)?